modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/AsyncSqlCursorImpl.java

```diff
@@ -71,6 +77,10 @@ public CompletionStage<BatchedResult<T>> requestNextAsync(int rows) {
                     t = t.getCause();
                 }
 
+                if (tx != null) {
```

Comment:
why we rollback the TX?

Reply:
we need to somehow handle tx here, this is exceptionally situation, i agree that probably not in all cases we need rollback, but this need to be discussed somehow and probably under differ issue.

Reply:
What are the cases when transaction must or shouldn't be rolled back?

One the one side. Here, as we already have a cursor, the query is already parsed and mapped to nodes.
So, we can't be sure whether any lock was obtained or not.

On the other side, it is a client API part and server errors are processed on the server-side.
Errors on the client side (e.g. disconnect and marshaller failures) can be handle by the user,
However, I doubt any error is recoverable here and TX will be likely aborted by the user.